### PR TITLE
add Dialog component

### PR DESCRIPTION
**Pull request by hcourthias**
## Checklist before requesting a review
- [x] Working on iOS
- [x] Working on Android
- [x] Integration tests added
- [x] Visual regressions screenshots are up to date
- [ ] Design validated

### CODE REVIEW

File: packages/sandbox/e2e/ios/screenshots/Dialog.png
Comment (hcourthias) [inline packages/sandbox/e2e/ios/screenshots/Dialog.png:1]: @ianrobertlambert The margin around the modal will not be their when using the component in the main project.

File: packages/streamline/src/components/dialog/dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        </Text>
+      </Box>
+      {children}
+      <Box
+        marginTop="xl"
+        flex={1}
+        flexDirection={
+          buttons.orientation === 'horizontal' ? 'row-reverse' : 'column'
+        }
+        style={{ marginBottom: insets.bottom }}
```
Comment (judastheo) [inline packages/streamline/src/components/dialog/dialog.tsx]: You can move this style and the `useSafeAreaInsets` in the `useStyles` part

File: packages/streamline/src/components/dialog/dialog.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+  return {
+    primary: {
+      ...(buttons.orientation === 'horizontal'
+        ? { marginLeft: spacing.xxs }
+        : {}),
+      ...(buttons.secondary ? { marginBottom: spacing.xxs } : {}),
+    },
+    secondary: {
+      ...(buttons.orientation === 'horizontal'
```
Comment (judastheo) [inline packages/streamline/src/components/dialog/dialog.tsx]: You check 3 times in the file `buttons.orientation === 'horizontal'`, you could have only one `isHorizontal` const.

File: packages/streamline/src/components/dialog/dialog.types.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+export interface DialogProps {
+  title: string;
+  icon?: IconsName;
+  buttons: {
+    orientation: 'horizontal' | 'vertical';
+    primary: ButtonProps & {
+      appearance: 'primary' | 'danger' | 'warning';
+    };
+    secondary?: ButtonProps & {
+      appearance: 'neutral' | 'secondary';
```
Comment (judastheo) [inline packages/streamline/src/components/dialog/dialog.types.tsx:13]: @hcourthias @ianrobertlambert Better safe than sorry, make sure you're aligned on possible appearance for each button.

File: packages/streamline/src/primitives/icon/icon.constants.ts [hunk 4]
```diff
@@ -2,10 +2,12 @@ export const ICON_SIZE = {
...
   regular: 16,
   large: 24,
+  xlarge: 48,
 };
 
 export const SPINNER_SIZE = {
   small: 16,
   regular: 24,
   large: 48,
+  xlarge: 48,
```
Comment (judastheo) [inline packages/streamline/src/primitives/icon/icon.constants.ts:12]: Discussed with @hcourthias , we may change it later but for now, it's just to match the icon different possibilities but should be of no use.

**Review by judastheo — CHANGES_REQUESTED** [review]
Nice!

**Review by judastheo — APPROVED** [review]

**Comment by ianrobertlambert** [comment]
Thanks @hcourthias - It's looking good! ✨ I have some questions and small feedback hopefully:

- [ ] The text content should be centered when the dialog uses an an icon.
- [ ] Does the `Bold` typography use a different typeface? I noticed that it was different on the screenshots, but not on the Expo environment? Is this issue specific to the `Dialog` component or is it a general `Typography` issue?
- [ ] Is there a way to test these in an overlay setting? It might provide a more accurate representation of their behavior on iOS/Android.
- [ ] Just to confirm, is the button(s) positioned 16px away from the edge of the `HomeIndicator` container on iOS/Android?
image

**Pull request merged**